src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -503,22 +508,27 @@ export class InputDatePicker
                 onCalciteInputInput={this.calciteInternalInputInputHandler}
                 onCalciteInternalInputBlur={this.calciteInternalInputBlurHandler}
                 onCalciteInternalInputFocus={this.startInputFocus}
+                onFocus={this.startEndInputFocus}
                 placeholder={this.localeData?.placeholder}
                 readOnly={readOnly}
+                role="combobox"
                 scale={this.scale}
                 type="text"
                 // eslint-disable-next-line react/jsx-sort-props
                 ref={this.setStartInput}
               />
+              {this.renderToggleIcon(this.open && this.focusedInput === "start")}
             </div>
             <div
               aria-hidden={toAriaBoolean(!this.open)}
               aria-label={messages.chooseDate}
+              aria-live="polite"
               aria-modal="true"
               class={{
                 [CSS.menu]: true,
                 [CSS.menuActive]: this.open
               }}
+              id="date-picker-dialog"
```

Comment:
I *think* if multiple dialogs are present on the page it's causing some confusion with screen readers. 

Can this `id` be more unique generating a guid from the [guid utils](https://github.com/Esri/calcite-components/blob/master/src/utils/guid.ts), associated with the name, similar to [`combobox`](https://github.com/Esri/calcite-components/blob/master/src/components/combobox/combobox.tsx#L1134)? 

NVDA picks up collapse, but JAWS isn't providing context. 😢

Reply:
Yeah these should probably be unique since we don't know how screen readers will deal with shadowRoot.

Reply:
And here I was, thinking shadow DOM would make this easier. 🥲 

Serious talk, thanks for pointing this out. I'll fix this right away. Can we add this to our a11y guidelines?